Updated configuration.md (parameter documentation)

susanmdouglas-aws opened the pull request:
Updated _internals/configuration.md to separate parameters into lists of parameters that can be modified and parameters that should not.  Also added information about default values and permissible parameter values.

Signed-off-by: susanmdouglas <susandou@amazon.com>

### Description
Updated _internals/configuration.md to separate parameters into lists of parameters that can be modified and parameters that should not.  Also added information about default values and permissible parameter values.
 
### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x ] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

rcv-aws reviewed (changes requested):
babelfishpg_tsql.use_antlr is scheduled to be removed from the source code
babelfishpg_tsql.enable_ownership_structure is scheduled to be removed from the source code

For babelfishpg_tsql.sql_dialect, please add a stronger warning like the following: "This GUC is for specific development use only. Any use of this GUC is not supported. Do not use this if you are a Babelfish user, as errors, malfunctions or functional incorrect behavior could result."

nasbyj commented on _internals/configuration.md:
```diff
@@ -193,391 +39,321 @@ Sets the first day of the week to a number from 1 through 7.
-Set database session property on TDS connections. What it means is that
-`babelfishpg_tsql.database_name` is set during the login process. An error will
-happen if the `database_name` is not set.
+You may come across these parameters in the source code, but you shouldn't change them:
```
Shouldn't we just list these out and not document them? Documenting them adds additional maintenance burden and invites people to mess with these parameters.

susanmdouglas-aws replied:
Done - I've removed the comments and reformatted this section as a bulleted list...

susanmdouglas-aws commented on _internals/configuration.md:
```diff
@@ -193,391 +39,321 @@ Sets the first day of the week to a number from 1 through 7.
+### `babelfishpg_tsql.language`
 
-### `babelfishpg_tsql.escape_hatch_rowguidcol_column`
+This parameter controls the T-SQL compatibility language option.
```
@rcv-aws @JimFinnerty @kuntalghosh @AvantikaDG 

I've dropped a note out requesting information about what babelfishpg_tsql.language option controls, how to set it, and what the valid values are... if any of you know, can you pass that along to me please or leave a note here?

susanmdouglas-aws replied:
I've asked multiple times on multiple Slack channels and no one has answered - I've also grepped through the repos and can't find the parameter in the source code.  At this point, I'm going to assume that this is a leftover from early development, and remove it...

susanmdouglas-aws replied:
Per our conversation, I've just added babelfishpg_tsql.language into the list of parameters that shouldn't be changed (in case it is available in an older/stable branch).

susanmdouglas-aws commented:
@rcv-aws I can't find a way to comment inline on your original message, but I've removed babelfishpg_tsql.use_antlr and babelfishpg_tsql.enable_ownership_structure and added babelfishpg_tsql.sql_dialect to the 'do not change' section (which is now a bulleted list per feedback from Jim).

rcv-aws commented on _internals/configuration.md:
```diff
@@ -193,391 +39,321 @@ Sets the first day of the week to a number from 1 through 7.
 TSQL triggers terminate if there is no transaction active at the end.
 
 
 ### `babelfishpg_tsql.dump_antlr_query_graph`
```
Yes, this should be considered unsupported. I wonder if we could remove this altogether?

rcv-aws commented on _internals/configuration.md:
```diff
@@ -193,391 +39,321 @@ Sets the first day of the week to a number from 1 through 7.
 `/tmp/antlr.dot`.
 
 
 ### `babelfishpg_tsql.enable_antlr_detailed_log`
```
this seems to be a developer GUC

susanmdouglas-aws replied:
Moved to list of parameters that shouldn't be changed...

susanmdouglas-aws commented:
> 

I've added:  

The following parameter is for specific development use only. Any use of `babelfishpg_tsql.sql_dialect` is not supported as errors, malfunctions or functional incorrect behavior could result:

- `babelfishpg_tsql.sql_dialect`

These parameters are scheduled for removal from the source code, and should not be modified:

- `babelfishpg_tsql.dump_antlr_query_graph`
- `babelfishpg_tsql.enable_antlr_detailed_log`
